docs/DeveloperGuide.md

```diff
@@ -247,7 +247,43 @@ The following activity diagram summarizes what happens when a user executes a ne
 
 _{more aspects and alternatives to be added}_
 
-
+### Command history feature
+#### Implementation
+The command history feature is facilitated by `CommandTextHistory`. It has a `commandHistory` arraylist that stores the history of commands entered by the user as strings 
+and a `currentCommandIndex` that is initialised to be `-1` to keep track of the current command index in the command history.
+The `CommandTextHistory` class implements the following operations:
+* `CommandTextHistory#addCommandHistory(String commandText)` — Adds a command text to the command history.
+* `CommandTextHistory#getPreviousCommand()` — Returns the previous command text in the command history.
+* `CommandTextHistory#getNextCommand()` — Returns the next command text in the command history.
+
+##### Getting previous/next command
+The `getPreviousCommand()` method will first check if `commandHistory` is empty by checking if `currentCommandIndex` is more than `0`. 
```

Comment:
Sorry, I am just curious that instead of checking the currentCommandIndex is more than 0, why we dont check the size of the commandHistory or check the currentCommandIndex is greater than -1? Since there are cases where the currentCommandIndex can point to the command at size - 2 as shown in the AddCommandHistory0.png, where the commandHistory is not empty.
Or maybe this approach aims to verify that there exists a previous command to be returned?